CONTRIBUTING_GO.md

```diff
@@ -44,3 +67,101 @@ import (
 
 If a package name isn't the same as its directory, use the explicit name.
 
+### Commit Testing
+
+All commits in a PR must pass tests. If they don't, fix the commits and/or [squash them](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Squashing-Commits) so that they do pass the tests. This should be done so that we can use git-bisect easily.
+
+We use CI tests which run when you push to your branch. To run the tests locally, you can run any of these: `make build`, `make install`, `make test`, `go test ./...`, depending on what you're looking to do. Generally `go test ./...` is your best bet.
```

Comment:
I would say `make test` and `make test_go_test` and `test_sharness_short`. `make test` runs both of them.